File: Sprint 2/front_end_sprint_2/shop-online/src/app/model/product.ts
import {TypeProduct} from './type-product';

export interface Product {
  id?: number,
  code?: string
  name?: string;
  price?: number;
  image?: string;
  isDeleted?: boolean;
  specification?: string;
  productDescription?: string;
  origin?: string;
  quantity?: number
  creationDate?: string
  typeProduct?: TypeProduct
}

File: module7/Sprint-2/back_end_sprint_2/shop-online/src/main/java/phuc/com/shoponline/model/product/Product.java
package phuc.com.shoponline.model.product;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Setter
@Getter
@RequiredArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(columnDefinition = "text")
    private String image;
    private String code;
    private String name;
    private Double price;
    @Column(columnDefinition = "bit(1) default 0")
    private Boolean isDeleted;
    @Column(columnDefinition = "text")
    private String specification;
    @Column(columnDefinition = "text")
    private String productDescription;
    private String origin;
    private Integer quantity;
    private Date creationDate;
    @ManyToOne
    @JoinColumn(name = "id_type_product",referencedColumnName = "id")
    private TypeProduct typeProduct;
}
